editor/src/components/navigator/navigator-item/component-picker.tsx

```diff
@@ -59,20 +59,97 @@ export function componentPickerOptionTestId(componentName: string, variant?: str
 }
 
 export const ComponentPicker = React.memo((props: ComponentPickerProps) => {
+  const { onItemClick } = props
+  const [selectedComponentKey, setSelectedComponentKey] = React.useState<string | null>(null)
   const [filter, setFilter] = React.useState<string>('')
+  const menuRef = React.useRef<HTMLDivElement | null>(null)
 
-  const allComponentsToShow: InsertMenuItemGroup[] = []
+  const changeFilter = useCallback(
+    (newFilter: string) => {
+      setFilter(newFilter)
+    },
+    [setFilter],
+  )
 
-  props.allComponents.forEach((c) => {
-    allComponentsToShow.push({
-      ...c,
-      options: c.options.filter((o) =>
-        o.label.toLocaleLowerCase().includes(filter.toLocaleLowerCase().trim()),
-      ),
+  const componentsToShow = useMemo(() => {
+    const allComponentsToShow: InsertMenuItemGroup[] = []
+    props.allComponents.forEach((c) => {
+      allComponentsToShow.push({
+        ...c,
+        options: c.options.filter((o) =>
+          o.label.toLocaleLowerCase().includes(filter.toLocaleLowerCase().trim()),
+        ),
+      })
     })
-  })
+    return allComponentsToShow
+  }, [props.allComponents, filter])
+
+  const flatComponentsToShow = useMemo(() => {
+    return componentsToShow.flatMap((c) => c.options)
+  }, [componentsToShow])
+
+  const highlightedComponentKey = useMemo(() => {
+    const firstOptionKey =
+      flatComponentsToShow.length > 0 ? flatComponentsToShow[0].value.key : null
+    if (selectedComponentKey == null) {
+      return firstOptionKey
+    }
+    // check if selectedComponentKey is still in the list
+    const found = flatComponentsToShow.some((c) => c.value.key === selectedComponentKey)
+    return found ? selectedComponentKey : firstOptionKey
+  }, [flatComponentsToShow, selectedComponentKey])
+
+  const onItemHover = useCallback(
```

Comment:
I think there might be a conflict here where when you leave your mouse over an element and then the container scrolls, this is re-triggered when it probably shouldn't be?

I am only pressing down arrow here, https://screenshot.click/02-16-jvmq1-huryn.mp4

Reply:
Interesting. Maybe I need to cancel pointer events until the scroll finishes